snuba/clickhouse/optimize/optimize.py

```diff
@@ -400,3 +414,25 @@ def is_busy_merging(clickhouse: ClickhousePool, database: str, table: str) -> bo
             return True
 
     return False
+
+
+def _hash_partition(partition_name: str) -> int:
+    sha1 = hashlib.sha1()
+    sha1.update(partition_name.encode())
+    return int(sha1.hexdigest(), 16)
+
+
+def should_optimize_partition_today(
+    partition_name: str, divide_partitions_count: int
+) -> bool:
+    """
+    Determines if a partition should be optimized today based on the partition name
+    and the current day of year.
+    """
+    if divide_partitions_count <= 0:
+        return True
+
+    return (
+        _hash_partition(partition_name) % divide_partitions_count
+        == datetime.now().timetuple().tm_yday % divide_partitions_count
```

Comment:
this could cause inconsistency at year boundaries (two even days in a row), if we wanted to change the input day to something like "days since epoch (unix timestamp / 86400)"